aws/terminator/application_services.py

```diff
@@ -150,6 +150,44 @@ def terminate(self):
         self.client.delete_file_system(FileSystemId=self.id)
 
 
+class KinesisStream(Terminator):
+    @staticmethod
+    def create(credentials):
+        def paginate_streams(client):
+            names = client.get_paginator('list_streams').paginate(
+                PaginationConfig={
+                    'PageSize': 100,
+                }
+            ).build_full_result()['StreamNames']
+
+            if not names:
+                return []
+
+            return [
+                client.describe_stream(StreamName=n)['StreamDescription'] for n in names
+            ]
+
+        return Terminator._create(credentials, KinesisStream, 'kinesis', paginate_streams)
+
+    @property
+    def created_time(self):
+        return self.instance['StreamCreationTimestamp']
+
+    @property
+    def id(self):
+        return self.instance['StreamName']
+
+    @property
+    def name(self):
+        return self.instance['StreamName']
+
+    def terminate(self):
```

Comment:
It looks like we can get a `ResourceInUseException` error deleting a stream that is already pending deletion:

https://docs.aws.amazon.com/kinesis/latest/APIReference/API_DeleteStream.html

> If the stream is in the ACTIVE state, you can delete it. After a DeleteStream request, the specified stream is in the DELETING state until Kinesis Data Streams completes the deletion.

I think we need to add an `ignore` function to ignore non-active streams.

Here's an example from another terminator class:

https://github.com/mattclay/aws-terminator/blob/42baefdff6b509555b5958d206b8fbf5e78de0a2/aws/terminator/compute.py#L58-L60

Reply:
@mattclay fixed